### PR TITLE
[RDY] Align gates_to_hell use tiles with footprint orientation

### PR DESCRIPTION
**Pull request by mugmuggy**
Die action calculates the use positions of the gates_to_hell using the wrong orientation for the offsets, they should align for the only passable tiles.

Spawn of a gates_to_hell can change the passable flags upon placement/removal since the footprint tile might be in a room. Can walk through a toilet up against a wall or prevent placing an object if the room is being constructed.

### CODE REVIEW

**Comment by mugmuggy** [comment]
Also added checks around the accessibility of the patient and grim use tile as it maybe possible the findIdleTile finds an isolated section of the map. Also adjusted the mirroring so the grim swipes whilst facing the patient. Also the direction for when the grim spawns on the use tile, ie scenarios where he can't walk

**Comment by TheCycoONE** [comment]
Change looks good to me.

File: CorsixTH/Lua/humanoid_actions/die.lua [hunk 1]
```diff
@@ -114,15 +114,20 @@ local action_die_tick_reaper; action_die_tick_reaper = permanent"action_die_tick
...
         humanoid.hole_use_tile_x = hole_x + spawn_scenario[7]
         humanoid.hole_use_tile_y = hole_y + spawn_scenario[8]
-        -- tile can't be in a room
-        if humanoid.world:getRoom(humanoid.hole_use_tile_x, humanoid.hole_use_tile_y) then
+        -- tile can't be in a room and must be accessible by the patient
+        if not humanoid.world:getPathDistance(humanoid.tile_x, humanoid.tile_y, humanoid.hole_use_tile_x, humanoid.hole_use_tile_y)
+            or humanoid.world:getRoom(humanoid.hole_use_tile_x, humanoid.hole_use_tile_y) then
+          return false
+        end
+        -- ensure grim won't be in a a room
```
Comment (lewri) [inline CorsixTH/Lua/humanoid_actions/die.lua:129]: Small typo here "a a"

Comment (TheCycoONE) [reply CorsixTH/Lua/humanoid_actions/die.lua:129]: @mugmuggy missed this one ^?

Comment (mugmuggy) [reply CorsixTH/Lua/humanoid_actions/die.lua]: My bad, I also observed something I need to check related to lava hole. Completely unrelated to this change, so I might see if I can replicate that and include that with this.

Comment (mugmuggy) [reply CorsixTH/Lua/humanoid_actions/die.lua]: I wasn't able to replicate the error, maybe it was a different save than I was thinking, playing around with a few different ones when it occured, but the error was around persistence in player_hospital line 272. So I guess progress with this anyway.

**Pull request merged**